### PR TITLE
Update plugin dev guide with ConfiguredAsCode rule

### PR DESCRIPTION
**Pull request by timja**
Here is our checklist for contributors. No hard requirement here, just a reminder

- [x] Please describe what you did
Add docs showing how to use our rule which automatically configures a plugin based on the yaml file

- [ ] Link to issue you're working on if there's a relevant one

- [ ] Did you provide a test-case to demonstrate feature actually works / issue is fixed ?

- [ ] Please also consider adding a line to [CHANGELOG.md](https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/CHANGELOG.md)

### CODE REVIEW

File: docs/PLUGINS.md [hunk 1]
```diff
@@ -233,11 +244,12 @@ some changes made to your plugin break this configuration model.
...
 
-About the later, in case you need to introduce some breaking changes, you can define a backward compatibility test case
+About the latter, in case you need to introduce some breaking changes, you can define a backward compatibility test case
 
-```yaml
-    @Test public void should_be_backward_compatible() throws Exception {
-        ConfigurationAsCode.get().configure(ConfigAsCodeTest.class.getResource("obsolete-configuration-as-code.yml").toString());
+```java
+    @Test 
+    @ConfiguredWithCode("obsolete-configuration-as-code.yml")
```
Comment (casz) [inline docs/PLUGINS.md]: wouldn't this require that you set JCasC's depreciation to be allowed instead of fail?

**Comment by casz** [comment]
actually not a chore, but documentation :smile:

File: docs/PLUGINS.md (hunk 1 above)
Comment (timja) [reply docs/PLUGINS.md]: its there 2 lines down after this block

Comment (casz) [reply docs/PLUGINS.md]: I must be blind :flushed:

**Comment by casz** [comment]
Feel free to to merge with your new found powers @timja

**Pull request merged**